src/solvers/anderson.jl

```diff
@@ -14,34 +12,40 @@ struct AndersonCache{Tx,To,Tdg,Tg,TQ,TR} <: AbstractSolverCache
     R::TR
 end
 
-function AndersonCache(df, ::Anderson{m}) where m
+function AndersonCache(df, m)
```

Comment:
I'm guessing this was done to have m be known to the compiler

Reply:
I think, the current implementation has the same type instability problem, it's just better hidden. In
https://github.com/JuliaNLSolvers/NLsolve.jl/blob/ddd16a5675703cd2e3fd528f0dae5012a50255eb/src/solvers/anderson.jl#L175
the creation of `Anderson{m}` is also type unstable.

As far as I understand, both these instabilities in the current and the new implementation should not be a problem, since the type instability should not affect the output type when calling `nlsolve` and after the first call of `Anderson{m}` and `AndersonCache(df, m)` everything can be inferred. To me it seems, the current implementation is just more complicated without providing any benefits.

Reply:
OK, I retract my objection then!